r-reproducibility.Rmd

```diff
@@ -0,0 +1,1211 @@
+poeple say "oh, don't bother with R packages, do everything in base R"... don't 
+listen to them. Do you build a computer from scratch everytime you want to do
+any type of work? Or a car when you want to drive somewhere? No, you don't. Make
+use of other people's hard work to make tools that simplify your life.
+- Split up your analyses steps into individual files (e.g. "model" file, "plot"
+file). Then `source` those files as needed or save the output in `data/` to use 
+it in other files.
+- Try not to have R scripts be too long (e.g. more than 500 lines of code). Keep
+script sizes as small as necessary and as specific as possible (have a single
+purpose). A script should have an end goal.
+
+## Readability
+
+There are two reasons we write code: to instruct the computer to do something and
+to record the steps we took to get a particular result for us or others to
+understand. For computers, *how* or *what* you write doesn't matter, as long as
+the code is correct. Computers don't need to *understand* the code. But humans
+do need to understand it. We need clear language and explicit meaning in order
+to understand what is going on. Humans write code, humans read code, and humans
+must maintain them and fix any errors. So, what you write and how you write it
+is extremely important.
+
+Like natural human languages, R has a relaxed approached to how R code is written.
+This has some nice advantages, but also some major disadvantages, notably that
+writing styles can be quite different across the world or even within one's own
+code. So, it's important to stick to some guidelines, for instance, as laid out
+by the tidyverse [style guide][r-style-guide].
+
+Some tips to use when working in R:
+
+- Think of writing code as if you are writing in a natural language like English.
+- Imagine that other people will be reading your code.
+- Keep the code clear, simple, and *readable*.
+- Stick to a *[style guide][r-style-guide]*.
+- Use full and descriptive words when typing and creating objects.
+- Use white space to separate concepts (empty lines between them, use spaces, and/or tabs).
+- Use RStudio R Script Sections (`"Code->Insert Section"` or `Ctrl-Shift-R`) to
+separate content in scripts.
+
+Even though R doesn't care about naming, spacing, and indenting, it really
+matters how your code looks. Coding is just like writing. Even though you may go
+through a brainstorming note-taking stage of writing, you eventually need to
+write correctly so others can understand, *and read*, what you are trying to
+say. Brainstorming and exploratory work is fine, but eventually you need to
+write code that will be legible. That's why using a [style guide][r-style-guide]
+is really important.
+
+### Exercise: Make the code more readable
+
+Using the [style guide][r-style-guide] found in the link, try to make these code
+more readable. The code below is in some way either *wrong* or incorrectly
```

Comment:
I think this exercise should focus on readability, not on finding/fixing errors - reduce cognitive load.